### PR TITLE
add `powerdns` collector (authoritative ns)

### PR DESCRIPTION
**Pull request by ilyam8**
Fixes: #500

**A lot of room for improvements**:
- powerdns authoritative server exposes much more metrics then we collect
- our grouping could be better

However, i think that we need not to do it now, but for sure and certain we need to do it later.

So i use same charts as python collector has (there are some little changes still).

### CODE REVIEW

File: config/go.d/powerdns.conf [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+#    Syntax:
+#      timeout: 1
+#
+#  - method
+#    HTTP request method.
+#    Syntax:
+#      method: GET
+#
+#  - body
+#    HTTP request method.
```
Comment (thiagoftsm) [inline config/go.d/powerdns.conf:105]: The `body` has the exact same description of `method`.

File: config/go.d/powerdns.conf [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+#    Syntax:
+#      body: '{fake: data}'
+#
+#  - headers
+#    HTTP request headers.
+#    Syntax:
+#      headers:
+#        X-API-Key: key
```
Comment (thiagoftsm) [inline config/go.d/powerdns.conf:113]: Should we keep this header considering that we have `api_key`?

File: config/go.d/powerdns.conf (hunk 1 above)
Comment (ilyam8) [reply config/go.d/powerdns.conf:105]: it is copy/paste, we have that typo in a lot of configs, i will fix it in another PR

File: config/go.d/powerdns.conf (hunk 2 above)
Comment (ilyam8) [reply config/go.d/powerdns.conf:113]: It is an example of `headers` option usage.

Some APIs use API keys for authorization, i thought that we could add `x_api_key` or `api_key` option to the https://github.com/netdata/go.d.plugin/tree/master/pkg/web#configuration-options

```yaml
x_api_key: secret
```
is equal to

```yaml
headers:
   X-API-KEY: secret
```

but easier/faster to set

Comment (ilyam8) [reply config/go.d/powerdns.conf:108]: ok, i removed the option, will add it in a separate PR.

**Review by thiagoftsm — APPROVED** [review]

**Pull request merged**